src/shared/bus-unit-util.c

```diff
@@ -390,6 +492,142 @@ int bus_append_unit_property_assignment(sd_bus_message *m, const char *assignmen
 
                 r = sd_bus_message_append(m, "v", "i", (int32_t) n);
 
+#ifdef HAVE_SECCOMP
+
+        } else if (streq(field, "SystemCallFilter")) {
+                bool whitelist;
+                const char *p;
+
+                r = sd_bus_message_open_container(m, 'v', "bas");
+                if (r < 0)
+                        return bus_log_create_error(r);
+
+                p = eq;
+                if (*p == '~') {
+                        whitelist = false;
+                        p++;
+                } else
+                        whitelist = true;
+
+                r = sd_bus_message_append_basic(m, 'b', &whitelist);
```

Comment:
pitfall: all our sd-bus APIs need "int"s instead of "bool"s when referencing booleans, as we don't use C99 "bool" in public APIs...